code-studio/src/js/initApp/project.js

```diff
@@ -132,6 +132,9 @@ var projects = module.exports = {
    * Whether this project uses Firebase for data storage.
    */
   useFirebase: function () {
+    if (!current) {
+      return;
+    }
```

Comment:
Good question -- this kind of defensive programming is a bit of a code smell. 

In one object-oriented vision of the world, this file would be a class which made sure `current` was initialized in the constructor and so you wouldn't ever be able to call `useFirebase` without `current` being defined (even if only as `{}`). However since the mechanism for ensuring that `current` exists is hard to know without an understanding that extends beyond this file, I'd rather keep it defensive for now.